lib/read.js

```diff
@@ -1,113 +1,77 @@
-var Juttle = require('juttle/lib/runtime').Juttle;
-var JuttleErrors = require('juttle/lib/errors');
-var JuttleMoment = require('juttle/lib/moment/juttle-moment');
-var Promise = require('bluebird');
-var google = require('googleapis');
-var _ = require('underscore');
-var Batchelor = require('batchelor');
-var FilterGmailCompiler = require('./filter-gmail-compiler');
+'use strict';
 
-var auth;
+/* global JuttleAdapterAPI */
+let AdapterRead = JuttleAdapterAPI.AdapterRead;
+let JuttleMoment = JuttleAdapterAPI.types.JuttleMoment;
 
-var Read = Juttle.proc.source.extend({
-    procName: 'read gmail',
+let Promise = require('bluebird');
+let google = require('googleapis');
+let _ = require('underscore');
+let Batchelor = require('batchelor');
+let FilterGmailCompiler = require('./filter-gmail-compiler');
 
-    initialize: function(options, params, location, program) {
-        this.logger.debug('intitialize', options, params);
-        this.gmail = google.gmail('v1');
+let auth;
 
-        var time_related_options = ['from', 'to', 'last'];
-        var allowed_options = time_related_options.concat(['raw']);
-        var unknown = _.difference(_.keys(options), allowed_options);
+class GmailRead extends AdapterRead {
 
-        if (unknown.length > 0) {
-            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR',
-                                     {proc: 'read gmail', option: unknown[0]});
-        }
+    periodicLiveRead() { return true;}
 
-        // One of 'from', 'to', or 'last' must be present.
-        var opts = _.intersection(_.keys(options), time_related_options);
-        if (opts.length === 0) {
-            throw this.compile_error('RT-MISSING-TIME-RANGE-ERROR');
-        }
+    defaultTimeRange() {
+        return {
+            from: this.params.now,
+            to: new JuttleMoment(Infinity)
+        };
+    }
 
-        // If 'from'/'to' are present, 'last' can not be present.
```

Comment:
I have a unit test for this which is passing (although I noticed I need to check for false negatives), and I see https://github.com/juttle/juttle/blob/master/lib/runtime/procs/source.js#L72.

With the addition of defaultTimeRange(), it's now acceptable to not specify any of -from/-to/-last, and I just updated the tests to reflect that.